### DISCUSSION TITLE
Preview is not available for buffers with python-mode.

### DISCUSSION THREAD

**Discussion by EugeneNeuron** [minad/consult#1146] [Q&A] [answered]
I noticed that `consult-imenu` provides different UIs in `emacs-lisp-mode` and `python-mode`. In the former, candidates are grouped by type (Types, Vars, Funcs) and preview is available, while in the latter, the type of the candidate is part of the candidate itself, without grouping, and no preview is offered.

I suspect this is because `python-mode` doesn't provide certain metadata for imenu, which prevents Consult from delivering such a detailed UI. I see that the `:state` of `consult--read` in `consult-imenu--select` disables preview for candidates without a marker.

Strictly speaking, this question is not about Consult itself, but about imenu. However, you likely understand the imenu internals needed for Consult to enable preview. I don't quite understand what's missing and where to start to enhance imenu metadata. Could you please provide a starting point?

**Comment by minad** [comment]
Preview should work with `python-mode` and `python-ts-mode` (works in my setup without any configuration afaik). For the organization, see `consult-imenu-config`.

**Reply by EugeneNeuron** [reply, marked as answer]
Thank you, I tried to turn on/off some parts of my config and found out that semantic-mode, being turned on, adds more candidates (variables, imports), but breaks consult live preview for some reason.
I commented out `(semantic-mode 1)` and that helped.
